examples/timer-interrupt-rtfm.rs

```diff
@@ -0,0 +1,99 @@
+//! Uses the timer interrupt to blink a led with different frequencies.
+//!
+//! This assumes that a LED is connected to pc13 as is the case on the blue pill board.
+//!
+//! Note: Without additional hardware, PC13 should not be used to drive an LED, see page 5.1.2 of
+//! the reference manaual for an explanation. This is not an issue on the blue pill.
+
+#![no_std]
+#![no_main]
+
+// you can put a breakpoint on `rust_begin_unwind` to catch panics
+extern crate panic_halt;
+
+use cortex_m::asm::wfi;
+use rtfm::app;
+
+use stm32f1xx_hal::{
+    prelude::*,
+    pac,
+    timer::{ Timer, Event },
+    gpio::{ gpioc::PC13, State, Output, PushPull },
+};
+
+#[app(device = stm32f1xx_hal::pac)]
+const APP: () = {
+
+    static mut LED: PC13<Output<PushPull>> = ();
+    static mut TIMER_HANDLER: Timer<pac::TIM1> = ();
+    static mut LED_STATE: bool = false;
+    
+    #[init]
+    fn init() -> init::LateResources {
+
+        // Take ownership over the raw flash and rcc devices and convert them into the corresponding
+        // HAL structs
+        let mut flash = device.FLASH.constrain();
+        let mut rcc = device.RCC.constrain();
+
+        // Freeze the configuration of all the clocks in the system and store the frozen frequencies
+        // in `clocks`
+        let clocks = rcc.cfgr.freeze(&mut flash.acr);
+
+        // Acquire the GPIOC peripheral
+        let mut gpioc = device.GPIOC.split(&mut rcc.apb2);
+
+        // Configure gpio C pin 13 as a push-pull output. The `crh` register is passed to the
+        // function in order to configure the port. For pins 0-7, crl should be passed instead
+        let led = gpioc.pc13.into_push_pull_output_with_state(&mut gpioc.crh, State::High);
+        // Configure the syst timer to trigger an update every second and enables interrupt
+        let mut timer = Timer::tim1(device.TIM1, 1.hz(), clocks, &mut rcc.apb2);
+        timer.listen(Event::Update);
+
+        // Init the static resources to use them later through RTFM
+        init::LateResources {
+            LED: led,
+            TIMER_HANDLER: timer,
+        }
+    }
+
+    #[idle]
+    fn idle() -> ! {
+
+        loop {
+            // Waits for interrupt
+            wfi();
```

Comment:
You can just not define an `idle()` function.

Reply:
Does it automatically call wfi() ?

Reply:
https://japaric.github.io/cortex-m-rtfm/book/en/by-example/app.html#idle

> When no idle function is declared, the runtime sets the SLEEPONEXIT bit and then sends the microcontroller to sleep after running init.